cli/lib/appdata.ts

```diff
@@ -20,9 +20,9 @@ const siteSchema = z
 
 const userDataSchema = z
 	.object( {
-		newSites: z.array( siteSchema ).optional(),
-		sites: z.array( siteSchema ).optional(),
-		snapshots: z.array( snapshotSchema ).optional(),
+		newSites: z.array( siteSchema ).default( () => [] ),
+		sites: z.array( siteSchema ).default( () => [] ),
+		snapshots: z.array( snapshotSchema ).default( () => [] ),
```

Comment:
This is not must-have, but it helps simplify the code elsewhere. It means that the appdata file will always contain empty array values for all these keys after the CLI touched it, but I didn't think that was really an issue.